crates/next-core/src/app_structure.rs

```diff
@@ -840,6 +840,83 @@ impl Issue for DuplicateParallelRouteIssue {
     }
 }
 
+#[turbo_tasks::value(shared)]
```

Comment:
Nit: remove `shared`, it's only needed if you're not exposing a constructor function from the module (but you are).

```suggestion
#[turbo_tasks::value]
```

https://turbopack-rust-docs.vercel.sh/rustdoc/turbo_tasks/attr.value.html